From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 17 Jul 2017 16:55:02 +0200
Subject: netfilter: nf_ct_expect: fix expect removal
Patch-mainline: submitted on 17th Jul 2017
References: bnc#1048935

Commit ec0e3f01114a ("netfilter: nf_ct_expect: Add
nf_ct_remove_expect()") introduced a helper nf_ct_remove_expect. It was
used over the code, but one location used a wrong variable and it
resulted in a crash in this call stack:
  -> nf_ct_expect_related_report
  -> nf_ct_remove_expect
  -> del_timer
  -> detach_if_pending

Switch to the proper variable.

Fixes: ec0e3f01114a
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Cc: Gao Feng <fgao@ikuai8.com>
Cc: Pablo Neira Ayuso <pablo@netfilter.org>
Cc: Jozsef Kadlecsik <kadlec@blackhole.kfki.hu>
Cc: Florian Westphal <fw@strlen.de>
Cc: <netfilter-devel@vger.kernel.org>
Cc: <coreteam@netfilter.org>
---
 net/netfilter/nf_conntrack_expect.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/netfilter/nf_conntrack_expect.c
+++ b/net/netfilter/nf_conntrack_expect.c
@@ -422,7 +422,7 @@ static inline int __nf_ct_expect_check(s
 	h = nf_ct_expect_dst_hash(net, &expect->tuple);
 	hlist_for_each_entry_safe(i, next, &nf_ct_expect_hash[h], hnode) {
 		if (expect_matches(i, expect)) {
-			if (nf_ct_remove_expect(expect))
+			if (nf_ct_remove_expect(i))
 				break;
 		} else if (expect_clash(i, expect)) {
 			ret = -EBUSY;
